test/scripts/integration/Run-IntegrationTests.ps1

```diff
@@ -34,8 +38,17 @@ $cmdArgs = @{
     Force = $true
 }
 $iterations = 2
+$expectedExitCode = 0
 & $functionTestScriptBlock
 
+"`n[sourcemod] Compile plugin via wrapper, bad plugin" | Write-Host
+$cmdArgs = @{
+    File = "$PSScriptRoot\..\..\mod\sourcemod\addons\sourcemod\scripting\plugin1_bad.sp"
+    Force = $true
+}
+$iterations = 2
+$expectedExitCode = 1
+& $functionTestScriptBlock
```

Comment:
The scriptblock [$functionTestScriptBlock](https://github.com/startersclan/Compile-SourceScript/pull/7/files#diff-56c3cff6bb7f2d2d84c4f3d644a5fb0aR9-R39) is simply a reusable piece of code that can be used in integration tests for testing PowerShell modules. It's just something I came up that can be used across multiple repos which isn't overly complicated I feel. Functions could be used too, though scoping of variables cannot be used in the same way as is used now with `script:` etc that makes it possible to prepare multiple parameters necessary for calling the tested function. Feel free to come up with one if you have the time to; but it had better be simple and as robust as it is now.

Reply:
@joeltimothyoh I recommend using Pester integration tests, but in a separate MR, instead of our own integration test implementation. So as to keep changes specific to the MR.

Reply:
Not a fan of an test-driven approach. I’m unsure I’ll actually bother writing such tests for a simple tool like this. Yes we can always open another PR for those changes.

Reply:
@joeltimothyoh it's not about test-driven, but more about taking a modular approach that results in more testable units of code.

Reply:
I'd steer clear from writing unit tests for something so fragile such as what's worked on in this PR. If functions are written from a low-level, it'd then be fine to have unit tests, but for changes such as those involved in this PR, there's much that we *don't* know and which involves invocation of multiple compilers and compiler wrappers, in all honesty, it's better to leave out the unit tests unless behavior of said scenarios has shown to be predictable and design of the module and its functions are mostly set in place.

For simplicity I think it's better to aim for real world uses first, and those simply involve integration tests. The module would not only be simpler to maintain, but can be refactored or revised accordingly quickly to fit the most practical of use cases.

Reply:
@joeltimothyoh sure. in the long run, for better ci processes esp since this is open source, having unit tests helps speed up contributors who can at least test majority of the code on their machine. Just a suggestion that could be implemented in the future